frame/election-provider-support/src/onchain.rs

```diff
@@ -103,6 +104,26 @@ impl<T: Config> ElectionProvider for OnChainSequentialPhragmen<T> {
 	}
 }
 
+impl<T: Config> ElectionProvider for OnChainSequentialPhragmen<T> {
+	type AccountId = T::AccountId;
+	type BlockNumber = T::BlockNumber;
+	type Error = Error;
+	type DataProvider = T::DataProvider;
+
+	fn elect() -> Result<Supports<T::AccountId>, Self::Error> {
+		Self::elect_with(None, None)
```

Comment:
Maybe a dq, but why is it safe to not put a limit on the number of voters and targets? Do we assume that if you are using `OnChainSequentialPhragmen` as the election provider then `DataProvider::{voters, target}` will always return values that won't OOM? If so we should document

Reply:
It is actually NOT safe, and you should NOT use this. I'll document it a bit better.

Reply:
Huh? If this is not safe, can you document something right here?

Reply:
It is documented at the type, where it is much more visible: 

```
/// Finally, the [`ElectionProvider`] implementation of this type does not impose any limits on the
/// number of voters and targets that are fetched. This could potentially make this unsuitable for
/// execution onchain. On the other hand, the [`InstantElectionProvider`] implementation does limit
/// these inputs.
///
/// It is advisable to use the former ([`ElectionProvider::elect`]) only at genesis, or for testing,
/// the latter [`InstantElectionProvider::instant_elect`] for onchain operations, with thoughtful
/// bounds.
```

Note that we never use this except at genesis, as recommended.